### DISCUSSION TITLE
Using README.md as directory description.

### DISCUSSION THREAD

**Discussion by equetzal** [doxygen/doxygen#11176] [Q&A]
So, I've been working on setting doxygen on a large project, and I have encountered an interesting situation where I'm not sure how to solve. This project has lot of directories, where each directory contains projects, libraries, components, etc. Each directory does have a `README.md` file that briefly describes the content of the directory. This means that my project has more than 50 readme files.

Usually, doxygen process all the `README.md` files as a "page", so it creates an entry in the related pages using the title of the H1 header. So, for Doxygen to process this markdown file as a directory description I've been adding `\dir` to the markdown file; something like this:

```md
\dir
# Some Title
Some description of this directory.
```

This properly archives the result of making the content of the readme file to be included in the directory description. However, it keeps the 'page' in the related pages with the title `README` and its content being blank. This result in over 50 blank pages called "README", which I cannot `\subpage` because there's no anchor/reference to it.

Other solution that I tried was to use the `USE_MDFILE_AS_MAINPAGE` set to `README.md`, this kind of 'merge' all the contents of the readme files into a single page, and because of the `\dir` directive, the result is not having the "README" pages in the related pages, while keeping the directory description. But the problem with this is that the actual mainpage gets discarded as multiple `\mainpage` directives are found, and my index.html is basically blank, which is also not ideal for the use that we have for the landing page. 

Is there a way to achieve the result of processing all the README files as directory-level descriptions, while keeping a specific markdown file for the index?

**Comment by albert-github** [comment]
I think I see what you mean but to be able to give a good answer and / or solution:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).
- Please also specify the **full** doxygen version used (`doxygen -v`).

**Comment by equetzal** [comment]
So I've attached the examples to you. I set `(readme)` in the titles the `README.md` files to differentiate them. 
[doxygen_example.zip](https://github.com/user-attachments/files/17336298/doxygen_example.zip)
```
doxygen -v 
1.9.7
```

So if I run doxygen as usual, **without** `USE_MDFILE_AS_MAINPAGE = README.md` and **no** README file having the `\dir` 
command this is the output I get:
- Directory without description
- A 'page' with the content of the README file
![image](https://github.com/user-attachments/assets/bea5b9ba-900f-43e6-a9d7-eb0910ac81a2)
![image](https://github.com/user-attachments/assets/84bd9c37-ba9b-4977-aba6-accd886744fd)

## If I add `\dir` at the beginning of the README files.
- The directory view uses the README content as the "Detailed Description"
- The 'pages' that used to have the title of the README are now titled "README" but are now blank
- The index is still intact, using `index.md` as is the one with the `\mainpage`
![image](https://github.com/user-attachments/assets/a2348247-85e8-4dea-99a6-36e03f14a41d)
![image](https://github.com/user-attachments/assets/1f745524-b2d4-46e9-a113-fd5086c91281)

## If I add `USE_MDFILE_AS_MAINPAGE = README.md` on top of the previous step
- The "README" pages disappear, while the directory view still uses the content as the "Detailed description"
- Now the index is gone, with warnings like (example), for each README file.
```txt
Search for main page...
D:/doxygen_example/example_readme_as_mainpage/doxygen_input/awesome_ui_component/README.md:1: warning: found more than one \mainpage comment block! (first occurrence: D:/doxygen_example/example_readme_as_mainpage/doxygen_input/astonishing_feature_component/README.md, line 1), Skipping current block!
```
![image](https://github.com/user-attachments/assets/51fa5abd-bd7d-44c0-9f9d-a4d688d3f734)
![image](https://github.com/user-attachments/assets/f8461d7d-62da-4783-939a-46dba97db2e4)

## Ideally..
The `README` 'pages' wouldn't appear in the related pages (Either because they are a README file or because their post-process content is empty), the content of the README with the`\dir` would go in each directory detailed description, and the single file with the `\mainpage` tag, should still be the index file.

**Comment by albert-github** [comment]
Converted to issue #11179
